### ISSUE TITLE
Generate a cleaner git history

### ISSUE THREAD

**Issue by dadoonet** [elastic/beat-generator#24]
I'm unsure where I should report this as I did not find which `Makefile` or other tool is actually doing the `git add.` when you run the first `make` on a new beat.

Whatever. By default all files are marked to be commited which means that the git history by default would have one single commit with all the files.
It's often adviced to add a single README file as the first commit so in case it's needed, you can always rewrite the history (only the first commit can not be rewritten).

I'd suggest to add files and commit them by default and generate a git log like this one:

```
3f9ae6d ‣ Add travis CI [3 hours ago by David Pilato]
0a22903 ‣ Add soundbeat skeleton [3 hours ago by David Pilato]
aaca10e ‣ Add empty contributing guide [3 hours ago by David Pilato]
7483ed0 ‣ Add gitignore and gitattributes [3 hours ago by David Pilato]
7b908eb ‣ Add License [3 hours ago by David Pilato]
24c73bc ‣ Initial commit [3 hours ago by David Pilato]
```

Last is the oldest commit.

So basically commit:

1. `README.md`
2. `LICENSE`
3. `.gitignore` and `.gitattributes`
4. `CONTRIBUTING.md`
5. Add all but `.travis.yml`
6. `.travis.yml`

WDYT?

**Comment by dadoonet** [comment]
Ha! I found the file: https://github.com/elastic/beat-generator/blob/master/%7B%7Bcookiecutter.beat%7D%7D/Makefile#L17

I can come with a PR if this makes sense.

**Comment by ruflin** [comment]
Definitive +1 on the first commit for the README. Not sure if we should squash the other 5 into 1 commit but have a good commit message. Your point here is probably that it should be very easy to remove for example the LICENSE from the history.

Initially I left out the hole commit but then added it as especially for new users it should be as easy as possible to create a beat and push it to Github. Should we perhaps make the "commit" a second command that is optional to run?

I'm fine with a PR with the changes above. It looks quite clean and it shows that already some stuff in the repo happened :-)

**Comment by dadoonet** [comment]
@ruflin I created a PR for this.
